examples/custom-session-validation/README.md

```diff
@@ -0,0 +1,120 @@
+## Feature Example - Invalidate Session Token
+
+This project demonstrates how to invalidate a session when a password is changed using password based authentication in your Keystone system.
+It builds on the [With Auth](../with-auth) example project.
+
+## Instructions
+
+To run this project, clone the Keystone repository locally, run `yarn` at the root of the repository then navigate to this directory and run:
+
+```shell
+yarn dev
+```
+
+This will start the Admin UI at [localhost:3000](http://localhost:3000).
+You can use the Admin UI to create items in your database.
+
+You can also access a GraphQL Playground at [localhost:3000/api/graphql](http://localhost:3000/api/graphql), which allows you to directly run GraphQL queries and mutations.
+
+## Features
+
+Based on the [With auth example](../with-auth/), this project demonstrates how to customise the kestone session configuration in order to invalidate a session when a user's passsword is changed.
+It uses the [`@keystone-6/auth`](https://keystonejs.com/docs/apis/auth) package, along with Keystone's [session management API](https://keystonejs.com/docs/apis/session), to add the following features:
+
+- Adds a hooks to the `password` field to set a timestamp field when the user changes their password
+- Changes the stateless session handling to set the time a session starts
+- Checks the session start time against when the user's password was last changed and returns no session if the password was changed _after_ the session started.
+
+### Added fields
+
+We add one new field, `passwordChangedAt`, to the `Person` list. Setting the `passwordChangedAt` field to hidden with access not allowed means this field will not be visible to the user. We then add a `resolveInput` hook on the `passwordChangedAt` field to set it to the current time whenever the password is changed.
+
+```typescript
+    email: text({ isIndexed: 'unique', validation: { isRequired: true } }),
+    password: password({
+        validation: { isRequired: true },
+      }),
+    passwordChangedAt: timestamp({
+        access: () => false,
+        hooks: {
+          resolveInput: ({ resolvedData }) => {
+            if (resolvedData.password) {
+              return new Date();
+            }
+            return;
```

Comment:
Can we add a comment to what this branch represents?

Reply:
Added comment in `./schema.ts` and expanded text above